### PR TITLE
Add CSV utility functions used in the Lab CSV import feature

### PR DESCRIPTION
**Pull request by cizl**
### Description

Adds a couple of small utility functions that will be used in the CSV import feature of the upcoming Lab release.

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "csv_utils.hpp"
+
+namespace CsvUtils {
+
+void CreateCsvFile(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
```
Comment (mpintaric55334) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:6]: Im not tagged as reviewer, but was "around", you dont need mgp::memory = memory;, that will be deprecated soon, DispatcherGuard is enough

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+  return;
+}
+
+void DeleteCsvFile(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
```
Comment (mpintaric55334) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:35]: Same comment about memory.

File: cpp/csv_utils_module/csv_utils_module.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include <mg_utils.hpp>
+#include <mgp.hpp>
+
+#include "algorithm/csv_utils.hpp"
+
+extern "C" int mgp_init_module(mgp_module *module, mgp_memory *memory) {
+  try {
+    mgp::memory = memory;
```
Comment (mpintaric55334) [inline cpp/csv_utils_module/csv_utils_module.cpp:8]: Same comment about memory, we use MemoryDispatcherGuard instead

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    const std::string_view filepath = arguments[0].ValueString();
+    const auto content = arguments[1].ValueString();
+    const auto isAppend = arguments[2].ValueBool();
+
+    std::ofstream fout;
+    fout.open(std::string(filepath), isAppend ? std::ofstream::app : std::ofstream::out);
+    fout << content << std::endl;
+    fout.close();
+
+    mgp::List return_list;
```
Comment (mpintaric55334) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:21]: Does this list do anything or used anywhere?

**Review by mpintaric55334 — CHANGES_REQUESTED** [review]
Wasn't tagged as reviewer, but was checking MAGE PR'S, so I left a few comments.

**Comment by antoniofilipovic** [comment]
@cizl is this ready for review or a work in progress? Because on PR you put WIP and on status ready :smile:

**Comment by cizl** [comment]
> @cizl is this ready for review or a work in progress? Because on PR you put WIP and on status ready 😄

Sorry about that, forgot to remove it from the title. :sweat_smile: 
Feel free to take a look!

**Comment by cizl** [comment]
> Wasn't tagged as reviewer, but was checking MAGE PR'S, so I left a few comments.

Nice! Thanks a bunch for taking a look :smile:

**Review by mpintaric55334 — APPROVED** [review]
LGTM! I just added empty lines to the end of the files and resolved the conflict in CMakeLists.txt so we dont't play PR ping pong over an empty line change.

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace CsvUtils {
+
+void CreateCsvFile(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard(memory);
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const std::string_view filepath = arguments[0].ValueString();
```
Comment (antoniofilipovic) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:11]: You can use `const auto` here, as your are using for other variables too

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    const auto content = arguments[1].ValueString();
+    const auto isAppend = arguments[2].ValueBool();
```
Comment (antoniofilipovic) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:13]: Use snake case for variable naming
```suggestion
    const auto is_append = arguments[2].ValueBool();
```

File: cpp/csv_utils_module/algorithm/csv_utils.cpp [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    std::ofstream fout;
+    fout.open(std::string(filepath), isAppend ? std::ofstream::app : std::ofstream::out);
+    fout << content << std::endl;
+    fout.close();
+
+    auto record = record_factory.NewRecord();
+    record.Insert(std::string(filepath).c_str(), filepath);
+
+    return;
```
Comment (antoniofilipovic) [inline cpp/csv_utils_module/algorithm/csv_utils.cpp:23]: Unnecessary return here as you have one below

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
All looks good, left few final comments

**Review by antoniofilipovic — APPROVED** [review]

**Pull request merged**